Formula/gecode.rb

```diff
@@ -12,8 +12,78 @@ class Gecode < Formula
     sha256 "a6cf500df618c42f0668bb227c090e6c1d3d3369c8b4537220d3deb78d5f8286" => :mountain_lion
   end
 
+  depends_on "qt5" => :optional
+
   def install
-    system "./configure", "--prefix=#{prefix}", "--disable-examples"
+    args = %W[
+      --prefix=#{prefix}
+      --disable-examples
+    ]
+    ENV.cxx11
+    if build.with? "qt5"
+      args << "--enable-qt"
+      ENV.append_path "PKG_CONFIG_PATH", "#{HOMEBREW_PREFIX}/opt/qt5/lib/pkgconfig"
+    else
+      args << "--disable-qt"
+    end
+    system "./configure", *args
     system "make", "install"
   end
+  
+  test do
+    (testpath/"test.cpp").write <<-EOS.undent
+      #include <gecode/driver.hh>
+      #include <gecode/int.hh>
+      #if defined(GECODE_HAS_QT) && defined(GECODE_HAS_GIST)
+      #include <QtGui/QtGui>
+      #if QT_VERSION >= 0x050000
+      #include <QtWidgets/QtWidgets>
+      #endif
+      #endif
+      using namespace Gecode;
+      class Test : public Script {
+      public:
+        IntVarArray v;
+        Test(const Options& o) : Script(o) {
+          v = IntVarArray(*this, 10, 0, 10);
+          distinct(*this, v);
+          branch(*this, v, INT_VAR_NONE(), INT_VAL_MIN());
+        }
+        Test(bool share, Test& s) : Script(share, s) {
+          v.update(*this, share, s.v);
+        }
+        virtual Space* copy(bool share) {
+          return new Test(share, *this);
+        }  
+        virtual void print(std::ostream& os) const {
+          os << v << std::endl;
+        }
+      };
+      
+      int main(int argc, char* argv[]) {
+        Options opt("Test");
+        opt.iterations(500);
+      #if defined(GECODE_HAS_QT) && defined(GECODE_HAS_GIST)
+        Gist::Print<Test> p("Print solution");
+        opt.inspect.click(&p);
+      #endif
+      
+        opt.parse(argc, argv);
+        Script::run<Test, DFS, Options>(opt);
+        return 0;
+      }
+    EOS
+  
+    args = %W[
+      -std=c++11
+      -I#{HOMEBREW_PREFIX}/opt/qt5/include
+      -I#{HOMEBREW_PREFIX}/include
+      -lgecodedriver -lgecodegist -lgecodesearch -lgecodeint -lgecodekernel -lgecodesupport
+      -L#{lib}
+      -o test
+    ]
+    system ENV.cxx, "test.cpp", *args
+    assert File.exist?("test")
+    system "./test | grep -q \"{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}\""
```

Comment:
Replace the above two lines with `assert_match "{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}", shell_output("./test")`